modules/exploits/windows/ftp/wing_ftp_admin_exec.rb

```diff
@@ -48,40 +53,53 @@ def initialize(info = {})
   end
 
   def check
-    res = send_request_cgi(
-      {
-        'uri'     =>  '/admin_login.html',
-        'method'  => 'GET'
-      })
+    @session_cookie = authenticate(datastore['USERNAME'], datastore['PASSWORD'])
 
-    if !res
-      fail_with(Failure::Unreachable, "#{peer} - Admin login page was unreachable.")
-    elsif res.code != 200
-      fail_with(Failure::NotFound, "#{peer} - Admin login page was not found.")
-    elsif res.body =~ /Wing FTP Server Administrator/ && res.body =~ /2003-2014 <b>wftpserver.com<\/b>/
-      return Exploit::CheckCode::Appears
-    end
+    ver = send_request_cgi(
+      'uri'       => '/admin_license.html',
+      'method'    => 'POST',
+      'cookie'    => 'UIDADMIN=' + datastore['UIDADMIN']
+    )
+
+    if ver.code != 200
+      fail_with(Failure::NotFound, "#{peer} - License page was not found.")
```

Comment:
`fail_with` should not be used inside `check` methods. (Admittedly about 30 modules in the framework do use `fail_with` inside `check`). A `CheckCode::` should be returned instead. ie; `return CheckCode::Safe`

Reply:
Fixed in commit 29627331cfeeb35d4a74d5128665e7e5d0f6d514.